packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -186,6 +189,39 @@ static BOOL ShouldSkipFocusZone(NSView *view)
 	return NO;
 }
 
+-(id)init
+{
+	self = [super init];
+	if (self != nil)
+	{
+		trackingArea = [[NSTrackingArea alloc] initWithRect:self.bounds options:NSTrackingActiveInActiveApp | NSTrackingMouseEnteredAndExited owner:self userInfo:nil];
+		[self addTrackingArea:trackingArea];
+	}
+	return self;
+}
+
+-(void)updateTrackingAreas
+{
+	[super updateTrackingAreas];
+	if (trackingArea != nil)
+	{
+		[self removeTrackingArea:trackingArea];
+	}
+
+	trackingArea = [[NSTrackingArea alloc] initWithRect:[self bounds] options:NSTrackingActiveInActiveApp | NSTrackingMouseEnteredAndExited owner:self userInfo:nil];
+	[self addTrackingArea:trackingArea];
+}
+
+- (void)mouseExited:(NSEvent *)event
+{
+	[super mouseExited:event];
+	NSView *viewWithFocus = GetFirstResponder([self window]);
+	if (viewWithFocus != nil)
+	{
+		[viewWithFocus resignFirstResponder];
+	}
+}
```

Comment:
This would affect all consumers of FocusZone, not just menu right? Would this negatively impact stuff like Radiogroup?

Reply:
That's right, I did a quick test on RadioGroup and Tabs before publishing the PR, didn't find anything weird... will do some more testing and report back

Reply:
just to be sure, lets make sure we test various edge cases we have been fixing over the years in focus zone so we aren't regressing anything.

Reply:
Debugged Tabs and RadioGroup in Xcode, I saw that `[viewWithFocus resignFirstResponder];` does get called on mouseExited. I think this is fine because it only removes the highlight, not the focus ring that those controls use.

source: https://developer.apple.com/documentation/uikit/uiresponder/1621097-resignfirstresponder

Reply:
that link is for uikit though. not app kit

Reply:
that is a good point. in the radio group, if it removes the highlight? what is focused? then?

Reply:
Scratch all that, a much simpler fix is to set focus on the menu container when onMouseLeave is called. Sorry for the noise :/